pkg/server/server.go

```diff
@@ -1126,21 +1115,8 @@ func New(ctx context.Context, config *lepconfig.Config, endpoint string, cliUID
 						continue
 					}
 
-					if cc, exists := lepconfig.DefaultContainerdComponent(ctx); exists {
-						ccfg := containerd_pod.Config{Query: defaultQueryCfg}
-						if cc != nil {
-							parsed, err := containerd_pod.ParseConfig(cc, dbRW, dbRO)
-							if err != nil {
-								log.Logger.Errorw("failed to parse component %s config: %w", name, err)
-								continue
-							}
-							ccfg = *parsed
-						}
-						if err := ccfg.Validate(); err != nil {
-							log.Logger.Errorw("failed to validate component %s config: %w", name, err)
-							continue
-						}
-						componentsToAdd = append(componentsToAdd, containerd_pod.New(ctx, ccfg))
+					if exists := containerd_pod.CheckContainerdRunning(ctx); exists {
```

Comment:
This code runs in a loop. We should check if the containerd component exists in componentsToAdd.. And the weird thing is that the loop is an array of component names. There should be a common factory, or simply iterate over an array of factories.  The pseudo code should be similar to the following
```go
for _, factoryFn := range []func(ctx) components.Component {
           containerd_pod.New, 
           docker_container.New, 
           k8s_pod.New,
} {
    // move these check such containerd_pod.CheckContainerdRunning into their factory fn
     component := factoryFn(ctx)
     componentsToAdd = append(componentsToAdd, component)
}
```

Reply:
@eahydra Yes, let us address in the following PR https://github.com/leptonai/gpud/issues/494, since we need changes in those 3 components